# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/wpf-gpu-raster/src/aarasterizer.rs
# Commit: 5762b27d2cc8
# Full Hash: 5762b27d2cc82e6c7f3f0d37bc4bcfdb130f9220
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-06-21 04:00:08
# Description:
#   Bug 1839459 - Elide empty closed sub-paths instead of emitting a start point. r=jrmuizel
#   
#   There is a corner case that triggers an assert in wpf-gpu-raster if an empty sub-path
#   starts off a path. Just avoid the problem by eliding the sub-path, but at least handle
#   close correctly as per the Canvas2D closePath spec:
# ==============================================================================

diff -r caa96b0874f4 -r 5762b27d2cc8 third_party/rust/wpf-gpu-raster/src/aarasterizer.rs
--- a/third_party/rust/wpf-gpu-raster/src/aarasterizer.rs	Tue Jun 20 18:01:09 2023 +0000
+++ b/third_party/rust/wpf-gpu-raster/src/aarasterizer.rs	Tue Jun 20 18:04:20 2023 +0000
@@ -1033,15 +1033,13 @@
         // Advance to the first point after the 'start' point:
         count -= 1;
         if (count == 0) {
-            //TraceTag!((tagMILWarning, "Path ended after start-path"));
-            //return (false);
-            return (true);
+            TraceTag!((tagMILWarning, "Path ended after start-path"));
+            return (false);
         }
 
         if ((types[1] & PathPointTypePathTypeMask) == PathPointTypeStart) {
-            //TraceTag!((tagMILWarning, "Can't have a start followed by a start!"));
-            //return (false);
-            return (true);
+            TraceTag!((tagMILWarning, "Can't have a start followed by a start!"));
+            return (false);
         }
 
         // Process runs of lines and Bezier curves: